modules/custom/dkan_data/src/Reference/Dereferencer.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+namespace Drupal\dkan_data\Reference;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\EntityTypeManager;
+use stdClass;
+
+/**
+ * Dereferencer.
+ */
+class Dereferencer {
+  use HelperTrait;
+
+  /**
+   * Indicates that dereferencing outputs data, the default case.
+   *
+   * @var int
+   */
+  const DEREFERENCE_OUTPUT_DEFAULT = 0;
+
+  /**
+   * Indicates that dereferencing outputs both the data and its uuid.
+   *
+   * @var int
+   */
+  const DEREFERENCE_OUTPUT_REFERENCE_IDS = 2;
```

Comment:
This `2` is a remnant from when we had three (de-)referencing methods. I must admit it bugs me a little that our values are 0 and 2. Seems a bit off, but if you're okay with it, I'm good.